extensions/resource-deployment/src/ui/notebookWizard/notebookWizardPage.ts

```diff
@@ -7,43 +7,53 @@ import { EOL } from 'os';
 import * as vscode from 'vscode';
 import * as nls from 'vscode-nls';
 import { NotebookWizardPageInfo } from '../../interfaces';
-import { initializeWizardPage, InputComponents, InputComponent, setModelValues, Validator } from '../modelViewUtils';
+import { initializeWizardPage, InputComponents, InputComponent, setModelValues, Validator, getTextComponent } from '../modelViewUtils';
 import { WizardPageBase } from '../wizardPageBase';
 import { NotebookWizard } from './notebookWizard';
 
 const localize = nls.loadMessageBundle();
 
 export class NotebookWizardPage extends WizardPageBase<NotebookWizard> {
 	private inputComponents: InputComponents = {};
+	private inputComponentOriginalValues!: Map<string, string>;
```

Comment:
hmm, i am still not sure why you want to keep the original values? for summary page, on page enter event, you just set the text components to the latest values, am i missing something?

Reply:
the key of the inputComponents is the variable name, isn't that enough?

Reply:
Yeah I don't understand why you need this mapping - the input variable name shouldn't be changing...

Reply:
The thing is that the template is stored in TextComponent.Value which changes and the original template is lost when $(variableNames) gets interpolated. 

So the evaluated_text can have something like: "This is a resource in Resourgroup:$(VAR_ResourceGroup) in subscription: $(VAR_Subscription)", specified as defaultValue. This defaultValue get put into the TextComponent' Value field as it is so the TextComponent.Value is: "This is a resource in Resourgroup:$(VAR_ResourceGroup) in subscription: $(VAR_Subscription)".

Now when this page gets loaded if the VAR_ResourceGroup was set as "RG1" and VAR_Subscription was set as "S1" then we can interpolate and change the value of TextComponent.Value to "This is a resource in Resourgroup:RG1 in subscription: S1". So everything works so far. But if the user now goes back and changes VAR_ResourceGroup to "RG2" when we load the said page then we have load the original "This is a resource in Resourgroup:$(VAR_ResourceGroup) in subscription: $(VAR_Subscription)". string and we can no longer replace the RG1 with RG2 to show them the correct value. Saving the original TextComponent.value in that maps allows to do the variable value substitution as many times as needed.

Reply:
Alan and I chatted online, so what we discussed was that instead of saving just the defaultValue of that field, we are going to save the whole FieldInfo object in the InputComponents array itself which can be used for this evaluation